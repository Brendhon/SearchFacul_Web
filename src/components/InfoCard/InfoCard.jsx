import React from 'react'
import './InfoCard.css'

import { FiPhone, FiMail, FiClock, FiMap, FiMapPin } from 'react-icons/fi'
import { FaStar } from 'react-icons/fa'

const InfoCard = props =>

    <div className="info-container">

        {/* Cabeçalho do info */}
        <header className="info-top">

            <strong className="info-top-title">{props.ies}</strong>
            <p className="info-top-subtitle">{props.category}</p>

        </header>

        {/* Conteúdo do Cartão */}
        <div className="info-content">

            <h5 className="info-content-title"> {props.name}</h5>

            <div className="info-content-subtitle">

                <p>{props.titration}</p>

                <p>{props.modality}</p>

                <p>{props.period}</p>

                <p className="align-items"><FiClock className="info-content-data-icon" />{props.duration} anos</p>

                {/* Mostrar pontuação caso ela exista*/}
                {props.score ?
                    <p className="align-items">{props.score} <FaStar className="info-content-data-icon" /></p> :
                    <div />
                }

            </div>

            <p className="info-content-description">{props.description}</p>

            <div className="info-content-data">

                <fieldset className="info-content-data-contact">

                    <legend>Contato</legend>

                    <p className="align-items"><FiMail className="info-content-data-icon" /> {props.email}</p>
                    <p className="align-items"><FiPhone className="info-content-data-icon" /> {props.telephone}</p>

                </fieldset>

                <fieldset className="info-content-data-address">

                    <legend>Localização</legend>

                    <p className="align-items"><FiMap className="info-content-data-icon" />{props.city} - {props.uf}</p>
                    <p className="align-items"><FiMapPin className="info-content-data-icon" />{props.address}</p>

                </fieldset>

            </div>

        </div>

        {/* Rodapé do Cartão */}
        <footer className="info-bottom">


            {/* Se a universidade possuir um site cadastrado ele ira aparecer */}
            {props.site ?

                // eslint-disable-next-line 
                <a className="a-link" target="_blank"
                    style={{ fontSize: 17, fontWeight: 600 }}
                    href={props.site}>Entre no site para saber mais</a> :
                <div />
            }

        </footer>

    </div >

export default InfoCard 