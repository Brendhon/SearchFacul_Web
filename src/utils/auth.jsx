export const isAuthenticated = _ => localStorage.getItem('authorization')