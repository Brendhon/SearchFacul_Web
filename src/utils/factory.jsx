const courseFactory = data => {
    return {
        name: data.name || "",
        description: data.description || "",
        duration: data.duration || "",
        titration: data.titration || "",
        modality: data.modality || "",
        period: data.period || "",
        score: data.score || ""
    }
}

const universityFactory = data => {
    return {
        IES: data.IES || "",
        telephone: data.telephone  || "",
        email: data.email || "",
        uf: data.uf || "",
        city: data.city || "",
        address: data.address || "",
        category: data.category || "",
        site: data.site || ""
    }
}

export { courseFactory, universityFactory }