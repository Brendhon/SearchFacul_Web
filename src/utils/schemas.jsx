import * as yup from 'yup'

const courseSchema = yup.object().shape({

    name: yup.string().required('Informe um nome'),
    duration: yup.string().length(1, 'Apenas um número inteiro').matches(/[0-9]+/gi, "Apenas números inteiros")
        .required('Informe a duração'),
    score: yup.string()
        .length(1, 'Apenas um número inteiro')
        .matches(/[0-5]+/gi, "Apenas números de 0 - 5"),
    description: yup.string().trim().required('Informe uma descrição'),
    titration: yup.string().required('Escolha uma titulação'),
    modality: yup.string().required('Escolha uma modalidade'),
    period: yup.string().required('Escolha um período'),

})

const updateCourseSchema = yup.object().shape({

    duration: yup.string().length(1, 'Apenas um número inteiro').matches(/[0-9]+/gi, "Apenas números inteiros"),
    score: yup.string()
        .length(1, 'Apenas um número inteiro')
        .matches(/[0-5]+/gi, "Apenas números de 0 - 5"),

})

const universitySchema = yup.object().shape({

    IES: yup.string().trim().required('Informe um IES da faculdade'),

    telephone: yup.string()
        .matches(/^[0-9]{8,11}$/, "Insira um número de telefone válido")
        .required('Informe um telefone para contato'),

    email: yup.string()
        .email('Deve ser um formato válido')
        .required('Informe um email para contato'),

    uf: yup.string()
        .matches(/^[A-Z]{2,2}$/, "Apenas letras maiúsculas")
        .required('Informe o UF do seu estado'),

    password: yup.string().trim().required('Informe uma senha'),

    confirmPassword: yup.string().oneOf([yup.ref('password'), null], 'As senhas não coincidem'),
    city: yup.string().trim().required('Informe a cidade'),
    address: yup.string().trim().required('Informe um endereço de referência'),
    site: yup.string().url('Deve estar no formato de Url'),
    category: yup.string().required('Uma categoria deve ser escolhida'),

})

const updateUniversitySchema = yup.object().shape({

    telephone: yup.string()
        .matches(/^[0-9]{8,11}$/, "Insira um número de telefone válido"),

    email: yup.string()
        .email('Deve ser um formato válido'),

    uf: yup.string()
        .matches(/^[A-Z]{2,2}$/, "Apenas letras maiúsculas"),

    password: yup.string(),

    confirmPassword: yup.string().oneOf([yup.ref('password'), null], 'As senhas não coincidem'),
    site: yup.string().url('Deve estar no formato de Url'),

})

const loginSchema = yup.object().shape({

    email: yup.string()
        .email('Deve ser um formato válido')
        .required('Informe um email'),

    password: yup.string().trim().required('Informe uma senha'),

})

const searchSchema = yup.object().shape({

    search: yup.string().required('Informe um conteúdo para a pesquisa'),

})

export {
    courseSchema,
    updateCourseSchema,
    universitySchema,
    updateUniversitySchema,
    loginSchema,
    searchSchema
}