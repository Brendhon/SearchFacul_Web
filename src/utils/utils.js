const filter = data => {
    for (let prop in data)
        if (!data[prop].trim())
            delete data[prop]
}

const match = (password1, password2) => {
    if (password1 === password2)
        return true
    else
        return false
}

const getError = error => {
    if (error.response.data.validation) {
        return `Dado incorreto: ${error.response.data.validation.body.keys[0]}`
    }
    else
        return error.response.data.message
}

module.exports = { filter, match, getError}
