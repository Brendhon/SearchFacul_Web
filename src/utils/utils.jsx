const getError = error => {
    if (error.response.data.validation)
        return `Dado incorreto: ${error.response.data.validation.body.keys[0]}`
    else
        return error.response.data.message
}

const removeEmptyData = data => {
    for (let prop in data) {

        if (typeof data[prop] === "number") {
            if (!data[prop])
                delete data[prop]
        }

        if (typeof data[prop] === "string") {
            if (!data[prop].trim())
                delete data[prop]
        }
    }
}

export { getError, removeEmptyData }