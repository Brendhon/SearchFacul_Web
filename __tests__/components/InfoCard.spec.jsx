import React from 'react'
import TestRenderer from 'react-test-renderer'

import * as CONSTANTS from '../utils/constants'
import InfoCard from '../../src/components/InfoCard/InfoCard'

describe("InfoCard", () => {

    // Atributos
    let data

    // Antes de todos os testes iniciar a variável
    beforeAll(() => {
        data = CONSTANTS.universityAndCourseData
    })

    it("Should be able to render", () => {

        // Testar a renderização do componente
        const component = TestRenderer.create(<InfoCard {...data} />).toJSON()

        // Comparando os Snapshots para verificar se teve alteração no component
        expect(component).toMatchSnapshot()
    })
})