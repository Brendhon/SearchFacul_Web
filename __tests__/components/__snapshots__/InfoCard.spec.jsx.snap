// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard Should be able to render 1`] = `
<div
  className="info-container"
>
  <header
    className="info-top"
  >
    <strong
      className="info-top-title"
    >
      Unifei
    </strong>
    <p
      className="info-top-subtitle"
    >
      pública
    </p>
  </header>
  <div
    className="info-content"
  >
    <h5
      className="info-content-title"
    >
       
      Engenharia de computação
    </h5>
    <div
      className="info-content-subtitle"
    >
      <p>
        Bacharelado
      </p>
      <p>
        Presencial
      </p>
      <p>
        Integral
      </p>
      <p
        className="align-items"
      >
        <svg
          className="info-content-data-icon"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <polyline
            points="12 6 12 12 16 14"
          />
        </svg>
        5
         anos
      </p>
      <p
        className="align-items"
      >
        4
         
        <svg
          className="info-content-data-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
          />
        </svg>
      </p>
    </div>
    <p
      className="info-content-description"
    >
      Melhor Curso
    </p>
    <div
      className="info-content-data"
    >
      <fieldset
        className="info-content-data-contact"
      >
        <legend>
          Contato
        </legend>
        <p
          className="align-items"
        >
          <svg
            className="info-content-data-icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
            />
            <polyline
              points="22,6 12,13 2,6"
            />
          </svg>
           
          guilherme@hotmail.com
        </p>
        <p
          className="align-items"
        >
          <svg
            className="info-content-data-icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M22 16.92v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"
            />
          </svg>
           
          34546789
        </p>
      </fieldset>
      <fieldset
        className="info-content-data-address"
      >
        <legend>
          Localização
        </legend>
        <p
          className="align-items"
        >
          <svg
            className="info-content-data-icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polygon
              points="1 6 1 22 8 18 16 22 23 18 23 2 16 6 8 2 1 6"
            />
            <line
              x1="8"
              x2="8"
              y1="2"
              y2="18"
            />
            <line
              x1="16"
              x2="16"
              y1="6"
              y2="22"
            />
          </svg>
          Itajubá
           - 
          MG
        </p>
        <p
          className="align-items"
        >
          <svg
            className="info-content-data-icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
            />
            <circle
              cx="12"
              cy="10"
              r="3"
            />
          </svg>
          Sei não
        </p>
      </fieldset>
    </div>
  </div>
  <footer
    className="info-bottom"
  >
    <a
      className="a-link"
      href="https://unifei.edu.br/"
      style={
        Object {
          "fontSize": 17,
          "fontWeight": 600,
        }
      }
      target="_blank"
    >
      Entre no site para saber mais
    </a>
  </footer>
</div>
`;
